tests/tests/test_preauth.py

```diff
@@ -205,10 +216,30 @@ def __wait_for_keygen():
             try:
                 run('stat {}'.format(Client.PRIV_KEY))
             except:
-                time.sleep(1)
-                sleepsec += 1
+                time.sleep(10)
+                sleepsec += 10
                 logging.info("waiting for key gen, sleepsec: {}".format(sleepsec))
             else:
+                time.sleep(5)
                 break
 
         assert sleepsec <= Client.KEYGEN_TIMEOUT, "timeout for key generation exceeded"
+
+    @staticmethod
+    def get_accepted_devices():
```

Comment:
the class is specifically for interacting with the client via ssh. this method should be added to `admission` perhaps? or as a helper to the test class. definitely not here.